pkg/operator/ceph/pool/controller.go

```diff
@@ -170,6 +170,8 @@ func (r *ReconcileCephBlockPool) reconcile(request reconcile.Request) (reconcile
 	if cephBlockPool.Status == nil {
 		updateStatus(r.client, request.NamespacedName, cephv1.ConditionProgressing, nil)
 	}
+	// update ObservedGeneration in status
+	cephBlockPool.Status.ObservedGeneration = cephBlockPool.ObjectMeta.Generation
```

Comment:
I don't think this line will actually cause the status to be updated unless I'm mistaken. I think we need to do something like note the observed generation here, then pass that to `updateStatus()` later on. Does that sound right?